### DISCUSSION TITLE
Our website ad on facebook is not opening in ios 14.7.1.

### DISCUSSION THREAD

**Discussion by hamitakti86** [vercel/next.js#29041] [Help]
we advertised on Facebook for our website, and we noticed that iPhone users using the current IOS (14.7.1) version are getting an error (An unexpected error has occured) in the Facebook browser when clicking on the ad and When I examined and clicked on the ads of other companies that advertise on facebook, I saw that the site url was working from the facebook browser without any problems. Our website is built using next js. Except for Iphone users using this IOS version (14.7.1) , we do not have a similar problem in any browser. We couldn't understand what happened.

I tried to upgrade all of the node packages in the project and then deploy it after that i realize error was changed like following;

Application error: a client-side exception has occurred (see the browser console for more information).

this seems like that facebook in-app-browser may not read some nextjs javascript syntax and i cannot open the fb browser console in mobile to see this error details.

I also defined the file pages/_error.js and add the following code: when i clicked the ad, it shows _error page warning, i added, i dont know how i see error  reason detailed of this issue. 

```
function Error({ statusCode }) {
  return (
    <p>
      {statusCode
        ? `An error ${statusCode} occurred on server`
        : 'An error occurred on client'}
    </p>
  )
}

Error.getInitialProps = ({ res, err }) => {
  const statusCode = res ? res.statusCode : err ? err.statusCode : 404
  return { statusCode }
}

export default Error
```

I wonder this might be a facebook browser's bug or i have to fix some mistakes in my code, i use eslint and i didnt any error and warning in my code. I will be happy if anyone with an idea can help.

**Comment by mckhames** [comment]
I have a similar issue with my own application. Somehow, someway, on iOS 14.7.1, the exact same error occurs on form-submission. Once the form is submitted, global state is set (with Zustand), and then the user is redirected to the "payment" page. Once on the payment page, the error is shown -- then, upon refreshing, the page loads just fine.

The really strange thing is that it is working on CHROME on iOS and NOT safari or duck-duck-go. It works in all situations on Chrome (PC) but do not have a Mac to check safari on desktop.

**Comment by hamitakti86** [comment]
This issue has resolved itself. I think facebook provided an update about it and now you can access the website from the ad link without any problems.
